### PR TITLE
backport: bitcoin#18736, #19379, #19296, #18775, #20188, #19558, #18445, #18744, #19067, #19065 (fuzzing harness backports: part 4)

### PR DESCRIPTION
**Pull request by kittywhiskers**

### CODE REVIEW

File: src/Makefile.test.include [hunk 1]
```diff
@@ -44,6 +44,7 @@ FUZZ_TARGETS = \
   test/fuzz/eval_script \
   test/fuzz/fee_rate \
   test/fuzz/fee_rate_deserialize \
+  test/fuzz/fees \
```
Comment (PastaPastaPasta) [inline src/Makefile.test.include:56]: 18736 why is this added here? did we skip a backport?

File: src/Makefile.test.include [hunk 2]
```diff
@@ -567,6 +571,12 @@ test_fuzz_fee_rate_deserialize_LDADD = $(FUZZ_SUITE_LD_COMMON)
 test_fuzz_fee_rate_deserialize_LDFLAGS = $(RELDFLAGS) $(AM_LDFLAGS) $(LIBTOOL_APP_LDFLAGS) $(LDFLAGS_WRAP_EXCEPTIONS)
 test_fuzz_fee_rate_deserialize_SOURCES = $(FUZZ_SUITE) test/fuzz/deserialize.cpp
 
+test_fuzz_fees_CPPFLAGS = $(AM_CPPFLAGS) $(BITCOIN_INCLUDES)
+test_fuzz_fees_CXXFLAGS = $(AM_CXXFLAGS) $(PIE_FLAGS)
+test_fuzz_fees_LDADD = $(FUZZ_SUITE_LD_COMMON)
+test_fuzz_fees_LDFLAGS = $(RELDFLAGS) $(AM_LDFLAGS) $(LIBTOOL_APP_LDFLAGS) $(LDFLAGS_WRAP_EXCEPTIONS)
+test_fuzz_fees_SOURCES = $(FUZZ_SUITE) test/fuzz/fees.cpp
+
```
Comment (PastaPastaPasta) [inline src/Makefile.test.include:649]: 18736 same as above

File: src/Makefile.test.include (hunk 1 above)
Comment (kittywhiskers) [reply src/Makefile.test.include:56]: The original `fees.cpp` was introduced in [#18565](https://github.com/bitcoin/bitcoin/pull/18565) through 44fb2a596b4a1aa70253c4145c35be6de68da22a which addressed logic that Dash doesn't have. `StringForFeeReason` exists in Dash and so we created the file on a different commit as opposed to the original file history ([source](https://github.com/practicalswift/bitcoin/commits/a4e3d13df6a6f48974f541de0b5b061e8078ba9a/src/test/fuzz/fees.cpp))

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

**Review by UdjinM6 — APPROVED** [review]
LGTM, utACK

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merge via merge commit
